### ISSUE TITLE
LaTeX coming from a cell does not render as MathJax

### ISSUE THREAD

**Issue by Juanlu001** [spatialaudio/nbsphinx#119]
See for instance:

http://docs.poliastro.space/en/latest/examples/Exploring%20the%20New%20Horizons%20launch.html

(source: http://nbviewer.jupyter.org/github/poliastro/poliastro/blob/119ebee/docs/source/examples/Exploring%20the%20New%20Horizons%20launch.ipynb)

nbviewer renders both the LaTeX coming from markdown cells and output cells, but nbsphinx does not. This might as well be a configuration issue on the Read the Docs side, but I decided to report this here first.

**Comment by Juanlu001** [comment]
...now that I see it, nbviewer has a special MathJax configuration:

```
tex2jax: {
              inlineMath: [ ['$','$'], ["\\(","\\)"] ],
              displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
              processEscapes: true,
              processEnvironments: true
            },
```

The LaTeX coming from output cells is enclosed in single dollar signs $. Perhaps this can be fixed in the rtd side? @humitos

**Comment by mgeier** [comment]
Thanks for the report @Juanlu001!
Can you please try to find out what's the difference between your example and http://nbsphinx.readthedocs.io/en/latest/code-cells.html#Math?

**Comment by Juanlu001** [comment]
Well, I think the difference is in the amount of dollar signs! `Quantity` objects from Astropy in particular:

https://github.com/astropy/astropy/blob/v2.0.1/astropy/units/quantity.py#L1295

So I guess this is not an issue in nbsphinx. I leave the decision to close this up to you :) In the meanwhile I will report this upstream. Thanks!

**Comment by mgeier** [comment]
For the record: it doesn't seem to work on Github either: https://github.com/poliastro/poliastro/blob/77df66f1a80a010d52471a8cb2231ef35bd6cdb9/docs/source/examples/Exploring%20the%20New%20Horizons%20launch.ipynb

**Comment by mgeier** [comment]
I didn't find a formal specification what is supposed to be the content of a `text/latex` output cell, neither in `notebook` nor in `nbformat`.

I guess it makes most sense to put a (single) block-level math environment in there.

OTOH, inline math might give better line breaks, see https://github.com/jupyter/nbconvert/issues/275.

I think it would be good if we had some official guidelines on this.

@Juanlu001 Would you mind asking the Jupyter people for guidelines?

**Comment by Juanlu001** [comment]
Sorry, just as a follow up: I tried to implement this change in Astropy in https://github.com/astropy/astropy/pull/6502, but I failed miserably. Will take care of that there, don't think this is a nbsphinx problem so I'm closing this. Thanks for your time!

**Comment by Juanlu001** [comment]
For the interested readers: restarted this effort in https://github.com/astropy/astropy/pull/8313.

**Comment by mgeier** [comment]
Thanks for the update @Juanlu001!

For the record: I slightly changed my opinion compared to what I said above (https://github.com/spatialaudio/nbsphinx/issues/119#issuecomment-326733405):

> I guess it makes most sense to put a (single) block-level math environment in there.

The problem with block-level environments is that they are not left-aligned when exported to LaTeX/PDF (either using `nbconvert` or `nbsphinx`). And they don't allow line breaks, see https://github.com/jupyter/nbconvert/issues/275.

I now think that it's best to enclose the math output in `$\displaystyle ...$`, which works for both MathJax and LaTeX.

I've already made this change in IPython: https://github.com/ipython/ipython/pull/11357 (available since release 7.1.0).

I'm currently working on getting this into SymPy, too: https://github.com/sympy/sympy/pull/15625.
